numba/targets/rangeobj.py

```diff
@@ -65,6 +65,25 @@ def range3_impl(context, builder, sig, args):
                                   range_state_type,
                                   state._getvalue())
 
+    @lower_builtin(len, range_state_type)
+    def range_len(context, builder, sig, args):
+        """
+        len(range)
+        """
+        (value,) = args
+        state = RangeState(context, builder, value)
+        res = RangeIter.from_range_state(context, builder, state)
+        return impl_ret_untracked(context, builder, int_type, builder.load(res.count))
+
+    @lower_builtin(len, range_iter_type)
+    def range_iter_len(context, builder, sig, args):
```

Comment:
What about listing the iterator and then getting the length? e.g. `len(list(iter(r)))`